scrapy/downloadermiddlewares/redirect.py

```diff
@@ -150,11 +154,17 @@ class RedirectMiddleware(BaseRedirectMiddleware):
     def process_response(
         self, request: Request, response: Response, spider: Spider | None = None
     ) -> Request | Response:
+        if hasattr(self.crawler.spider, "handle_httpstatus_list"):
+            warn_on_deprecated_spider_attribute(
+                "handle_httpstatus_list", "REDIRECT_ALLOWED_HTTP_CODES"
+            )
+
         if (
             request.meta.get("dont_redirect", False)
             or response.status
             in getattr(self.crawler.spider, "handle_httpstatus_list", [])
             or response.status in request.meta.get("handle_httpstatus_list", [])
+            or response.status in self.handle_httpstatus_list
```

Comment:
I wonder if we should handle this somewhat differently.

i.e., when both the setting and the spider attribute are defined, the spider attribute, even if deprecated, should be considered an override of the setting value, and the setting value should be ignored.

In fact, I wonder if we should also consider the meta value an override of both the setting and the spider attribute, and hence ignore those if the meta value is set. But I am fine with keeping that out of scope here.

Either way, we should make sure we clarify this in the docs and the release notes.